guides/hack/11-built-in-types/28-shapes.md

```diff
@@ -95,6 +95,19 @@ function takes_server(Server $s) {
 }
 ```
 
+While you can't instantiate these types directly, you can create a my_function_arguments_t by simply defining a shape that has all of the required fields. The typechecker will automatically infer if your shape can be accepted by code() or not.
```

Comment:
I think this is a bit unclear; perhaps:

Any shape value that has all of the required fields (and no undefined fields - unless the shape permits them) is considered a value of  type `Server`; the type is not specified when creating the value.